outlines/types/__init__.py

```diff
@@ -25,10 +29,35 @@ class Json:
 
     """
 
-    def __init__(self, definition: Union[str, dict, BaseModel]):
-        if isinstance(definition, type(BaseModel)):
-            definition = definition.model_json_schema()
-        if isinstance(definition, str):
-            definition = json.loads(definition)
+    definition: Union[str, dict]
 
-        self.definition = definition
+    def to_json_schema(self):
+        if isinstance(self.definition, str):
+            schema = json.loads(self.definition)
+        elif isinstance(self.definition, dict):
+            schema = self.definition
+        elif isinstance(self.definition, type(BaseModel)):
+            schema = self.definition.model_json_schema()
```

Comment:
We also need to support `TypedDict`. We may be able to do this [via Pydantic](https://docs.pydantic.dev/2.3/usage/types/dicts_mapping/).

Reply:
Done, that was an easy change.